xclim/cli.py

```diff
@@ -0,0 +1,239 @@
+# -*- coding: utf-8 -*-
+"""
+xclim command line interface module
+"""
+import inspect
+
+import click
+import xarray as xr
+from dask.diagnostics import ProgressBar
+
+import xclim as xc
+from xclim import atmos
+from xclim import land
+from xclim import seaIce
+
+
+xcmodules = {"atmos": atmos, "land": land, "seaIce": seaIce}
+
+
+def _get_indicator(indname):
+    if "." in indname:
+        modname, indname = indname.split(".")
+    else:
+        raise click.BadArgumentUsage(
+            f"Indicator name must include the module name (ex: atmos.tg_mean) (got {indname})"
+        )
+
+    if modname not in xcmodules or indname not in xcmodules[modname].__dict__:
+        raise click.BadArgumentUsage(
+            f"Indicator '{indname}' or module '{modname}' not found in xclim."
+        )
+
+    return xcmodules[modname].__dict__[indname]
+
+
+def _get_input(ctx):
+    arg = ctx.obj["input"]
+    if isinstance(arg, xr.Dataset):
+        return arg
+    elif isinstance(arg, tuple) or "*" in arg:
+        ctx.obj["xr_kwargs"].setdefault("combine", "by_coords")
+        ds = xr.open_mfdataset(arg, **ctx.obj["xr_kwargs"])
+    else:
+        ctx.obj["xr_kwargs"].pop("combine", None)
+        ds = xr.open_dataset(arg, **ctx.obj["xr_kwargs"])
+    ctx.obj["input"] = ds
+    return ds
+
+
+def _get_output(ctx):
+    if "ds_out" not in ctx.obj:
+        dsin = _get_input(ctx)
+        ctx.obj["ds_out"] = dsin.drop_vars(dsin.data_vars)
```

Comment:
I don't think this is the expected behavior. If you have a file with 100 variables, they're all going to get copied except the inputs, correct ?

Reply:
This copies all attributes and coordinate variables from the input to the output. I guess if you had 100 coordinates that might be too much, but that's a improbable case, no?

Reply:
Agreed, but even if there is a just of couple of extra unrelated variables in my input data, I'd be surprised to see them pop up in the output. Is suggest explicitly copying over global attributes that are identical for all input files.